backend/python/app/services/implementations/entity_service_mg.py

```diff
@@ -35,7 +45,28 @@ def create_entity(self, entity):
         return new_entity.to_serializable_dict()
 
     def update_entity(self, id, entity):
-        updated_entity = Entity.objects(id=id).modify(new=True, **entity.__dict__)
+        current_entity = Entity.objects.get(id=id)
+        if current_entity is None:
+            self.logger.error("Invalid id")
+            raise Exception("Invalid id")
+        file_name = current_entity.file_name
+        if entity.file:
+            if file_name:
+                self.file_storage_service.update_file(
+                    file_name, entity.file, entity.file.content_type
+                )
+            else:
+                file_name = str(uuid4())
+                self.file_storage_service.create_file(
+                    file_name, entity.file, entity.file.content_type
+                )
+            entity.__dict__.pop("file", None)
+        elif file_name:
+            self.file_storage_service.delete_file(file_name)
```

Comment:
Need to add:
```python
entity.__dict__.pop("file", None)
file_name = None
```

The second line results in a null `file_name` field in MongoDB, would be nice if we can remove the field entirely if there is no file. Not a priority though, feel free to just leave it.

Reply:
Oh `file_name = None` is a good catch, updated that.

Not passing in `file_name=None` to `Entity.objects(id=id).modify(...)` keeps the old filename. Also tried [this](https://github.com/MongoEngine/mongoengine/issues/2128) but it did not work either.

Also not sure if deleting the field is a better pattern than having it as null because the only way to do it in postgres is to have it as a nullable column.

Reply:
> Also not sure if deleting the field is a better pattern than having it as null because the only way to do it in postgres is to have it as a nullable column.

Yeah that's fair, let's leave it then 👍